samza-yarn/src/main/java/org/apache/samza/job/yarn/YarnContainerRunner.java

```diff
@@ -110,6 +111,7 @@ public void runContainer(String samzaContainerId, Container container, CommandBu
     log.info("Container ID {} using command {}", samzaContainerId, command);
 
     Map<String, String> env = getEscapedEnvironmentVariablesMap(cmdBuilder);
+    env.put(ShellCommandConfig.ENV_EXECUTION_ENV_CONTAINER_ID(), Util.envVarEscape(container.getId().toString()));
```

Comment:
Ah my bad. I got confused between Samza's containerId and yarn's container id. You are right. This is the right place to add it. We are passing all of AM container's environmental variables to the container's context and additionally, including the yarn container id.